src/hipchatbot.js

```diff
@@ -0,0 +1,198 @@
+      "key": "com.jeopardy.bot",
+      "links": {
+        "homepage": "https://github.com/kesne/jeopardy-bot",
+        "self": HIPCHAT_CALLBACK_URL + "/capabilities",
+      },
+      "capabilities": {
+        "hipchatApiConsumer": {
+          "scopes": [
+            "admin_room",
+            "send_notification",
+          ]
+        },
+        "installable": {
+          "callbackUrl": HIPCHAT_CALLBACK_URL + "/install",
+          "allowGlobal": false,
+          "allowRoom": true,
+        },
+      },
+    }
+
+    res.send(descriptor);
+  }
+
+  async install(req, res) {
+    const { capabilitiesUrl, oauthId, roomId, oauthSecret } = req.body;
+    this.app.hipchat.oauthId = oauthId;
+    this.app.hipchat.oauthSecret = oauthSecret;
+
+    let tokenUrl;
+    request(capabilitiesUrl, (err, res, body) => {
+      if (!err && res.statusCode === 200) {
+        tokenUrl = JSON.parse(body).capabilities.oauth2Provider.tokenUrl;
+      } else {
+        this.onError(err);
+      }
+    });
+
+    const { access_token, expires_in } = await this.getAccessToken(oauthId, oauthSecret, tokenUrl);
+    this.app.hipchat.accessToken.token = access_token;
+    this.app.hipchat.accessToken.expires = moment().add(expires_in, 'seconds');
+    await this.app.save();
+
+    res.sendStatus(200);
+  }
+
+  getAccessToken(oauthId, oauthSecret, tokenUrl = 'https://api.hipchat.com/v2/oauth/token') {
+    return new Promise((resolve, reject) => {
+      request({
+        method: 'POST',
+        url: tokenUrl,
+        auth: {
+          user: oauthId,
+          pass: oauthSecret,
+        },
+        form: {
+          grant_type: 'client_credentials',
+          scope: [
+            'admin_room',
+            'send_notification',
+          ],
+        },
+      }, (err, res, body) => {
+        if (!err) {
+          resolve(JSON.parse(body));
+        } else {
+          reject(err);
+        }
+      });
+    });
+  }
+
+  onMessage(req) {
+    this.validateToken(this.app.hipchat);
+    const { 
+      event: subtype,
+      item: { 
+        room: { name: channel_name, }, 
+        message: { 
+          message: text, 
+          from: { 
+            id: user_id, 
+            mention_name: user_name,
+          },
+        },
+      },
+    } = req.body;
+
+    const channel_id = (req.body.item.room.id).toString();
+    const timestamp = Date.parse(req.body.item.message.date) / 1000;
+    
+    const say = (message, url = '') => {
+      if (url) { message += '<br />' + '<img src="'+url+'" width="420" height="259" /><br />'; }
+      this.hipchatter.notify(channel_name, { message: message, token: this.app.apiToken }, this.onError.bind(this));
+    };
+
+    trebek(text, {
+      subtype,
+      channel_name,
+      channel_id,
+      timestamp,
+      user_id,
+      user_name,
+    }, say);
+  }
+
+  async registerWebhook(roomId, event) {
+    for (let webhookId of this.app.webhooks) {
+      const wh = await new Promise((resolve, reject) => {
```

Comment:
I found myself a little lost on some of the async/await and promise syntax while making some of these changes. So if you see anything that looks weird or wrong let me know